src/modules/emit_sound.rs

```diff
@@ -0,0 +1,219 @@
+                info.to,
+                entity,
+                info.channel,
+                sound,
+                // [0,255]
+                (info.volume * 255.).round() as i32,
+                info.attenuation,
+                info.flag,
+                info.pitch,
+            );
+        };
+    }
+}
+
+static BXT_EMIT_SOUND_DYNAMIC: Command = Command::new(
+    b"bxt_emit_sound_dynamic\0",
+    handler!(
+        "bxt_emit_sound_dynamic <sound> <channel> [volume] [from] [to] [attenuation] [flag] [pitch]
+
+Plays sound file directly from S_StartDynamicSound.",
+        emit_sound_dynamic as fn(_, _, _),
+        emit_sound_dynamic_full as fn(_, _)
+    ),
+);
+
+fn emit_sound_dynamic(marker: MainThreadMarker, sound: String, channel: i32) {
+    emit_sound_dynamic_full(marker, SoundInfo::new(sound, channel))
+}
+
+fn emit_sound_dynamic_full(marker: MainThreadMarker, info: SoundInfo) {
+    if let Some(player) = unsafe { engine::player_edict(marker) } {
+        let origin = unsafe {
+            if info.to == 0 {
+                // It does this to have the sound follow player's vieworg rather than origin.
+                *engine::listener_origin.get(marker)
+            } else {
+                let to = NonNull::new(player.as_ptr().add(info.to as usize)).unwrap();
+                to.as_ref().v.origin
+            }
+        };
+
+        let binding = CString::new(info.sound).unwrap();
+        let sound = binding.as_ptr();
+        let precache_result = unsafe { engine::S_PrecacheSound.get(marker)(sound) };
+
+        unsafe {
+            engine::S_StartDynamicSound.get(marker)(
+                info.from,
+                info.channel,
+                precache_result,
+                origin.as_ptr(),
+                info.volume,
+                info.attenuation,
+                info.flag,
+                info.pitch,
+            )
+        };
+    }
+}
+
+static BXT_EMIT_SOUND_STOP_SOUND_EXCEPT: Command = Command::new(
+    b"bxt_emit_sound_stop_sound_except\0",
+    handler!(
+        "bxt_emit_sound_stop_sound_except <list of channels seperated by space>.
+        
+Stops emitting sounds from every channel except given list of channels.",
+        stop_sound as fn(_, _)
+    ),
+);
+
+// It goes up to 14-15 something. I've seen up to 8 used so far.
+static MAX_CHANNELS: i32 = 10;
+
+fn stop_sound(marker: MainThreadMarker, channels: String) {
+    let mut list = vec![];
+    // There should be only 8 usable channels from SV_StartSounds.
+    // But SND_PickDynamicChannel says something very weird otherwise.
+    let mut list0 = Vec::from_iter(0..MAX_CHANNELS + 1);
+    let iter = channels.split_ascii_whitespace();
+
+    for channel in iter {
+        if let Ok(allow) = channel.parse::<i32>() {
+            list.push(allow);
+        }
+    }
+
+    // Eh.
+    list.sort();
+
+    for channel in list.iter().rev() {
+        let channel = (*channel).clamp(0i32, MAX_CHANNELS);
+        list0.remove(channel as usize);
```

Comment:
You could instead collect individual channels into a HashSet, then do `for channel in 0..=MAX_CHANNELS {}` where call StopSound if the HashSet doesn't contain the index.